### PR TITLE
Autofix with config

### PR DESCRIPTION
**Pull request by dani-santos-code**
### Description

Closes https://github.com/Shopify/kubeaudit/issues/264

##### Type of change

- [ ] Bug fix :bug:
- [x] New feature :sparkles:
- [ ] This change requires a documentation update :book:
- [ ] Breaking changes :warning:
##### How Has This Been Tested?

- [x] Manual tests

##### Checklist:

- [x] I have :tophat: my changes (A 🎩 specifically includes pulling down changes, setting them up, and manually testing the changed features and potential side effects to make sure nothing is broken)
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] The test coverage did not decrease
- [ ] I have signed the appropriate [Contributor License Agreement](https://cla.shopify.com/)

### CODE REVIEW

File: cmd/commands/autofix.go [hunk 1]
```diff
@@ -4,18 +4,30 @@ import (
...
+	kubeauditConfigFile string
 }
 
 func autofix(cmd *cobra.Command, args []string) {
-	report := getReport()
+	conf := loadKubeAuditConfigFromFile(autofixConfig.kubeauditConfigFile)
+
+	conf = setConfigFromFlags(cmd, conf)
+
+	allAuditors, err := all.Auditors(conf)
```
Comment (dani-santos-code) [inline cmd/commands/autofix.go:23]: should I just call it `auditors`? or `configEnabledAuditors`?

Comment (genevieveluyt) [reply cmd/commands/autofix.go:23]: Auditors makes sense to me! This code looks like it's based on the `all` command so I think we should change it there too!

File: cmd/commands/autofix.go [hunk 2]
```diff
@@ -35,20 +47,42 @@ func autofix(cmd *cobra.Command, args []string) {
 	}
 }
 
+func loadKubeAuditConfigFromFile(configFile string) config.KubeauditConfig {
```
Comment (genevieveluyt) [inline cmd/commands/autofix.go:50]: This looks like it's based on `loadConfigFromFile` from the `all` command. I think that function was meant to use `configFile` (the passed in parameter) instead of `auditAllConfig.configFile` (since the passed in parameter is not used anywhere in the function 🤦 ). I do like `loadKubeAuditConfigFromFile` better as a function name though. So I think what we should do is keep this function but replace `autofixConfig.kubeauditConfigFile` with `configFile`, then remove the `loadConfigFromFile` function, and have the auditAll command call this function instead. Wdyt?

File: docs/autofix.md [hunk 3]
```diff
@@ -263,3 +264,13 @@ To write the fixed manifest to a different file, use the `--outfile/-o` flag:
 ```
 kubeaudit autofix -f "manifest.yml" -o "fixed.yaml"
 ```
+
+### Using Custom Rules with Kubeaudit Config File
```
Comment (genevieveluyt) [inline docs/autofix.md:268]: Now that two commands support the kubeaudit config maybe we can move the documentation from the `all` command (https://github.com/Shopify/kubeaudit/blob/master/docs/all.md#kubeaudit-config) to the home page (https://github.com/Shopify/kubeaudit#configuration-file) and then link there from the `all` and `autofix` readmes?

**Review by genevieveluyt — COMMENTED** [review]
Awesome this is very clean!

File: cmd/commands/autofix.go (hunk 2 above)
Comment (dani-santos-code) [reply cmd/commands/autofix.go:50]: totally!! it makes sense! I also failed to notice `configFile` wasn't being used! 😂I will change it and also rename the function!

File: docs/autofix.md (hunk 3 above)
Comment (dani-santos-code) [reply docs/autofix.md:268]: good idea!!!

File: README.md [hunk 4]
```diff
@@ -216,7 +222,50 @@ Auditors can also be run individually.
...
+    # will be generated for containers which use an image without a tag
+    image: 'myimage:mytag'
+  limits:
+    # If no limits are specified and the 'limits' auditor is enabled, WARN results
+    # will be generated for containers which have no cpu or memory limits specified
+    cpu: '750m'
+    memory: '500m'
+```
+
+For more details about each auditor, including a description of the auditor-specific configuration in the config, see the [Auditor Docs](/README.md#auditors).
```
Comment (genevieveluyt) [inline README.md:266]:
```suggestion
For more details about each auditor, including a description of the auditor-specific configuration in the config, see the [Auditor Docs](#auditors).
```

File: fix.go [hunk 5]
```diff
@@ -1,135 +0,0 @@
-package kubeaudit
```
Comment (genevieveluyt) [inline fix.go:1]: Why is this file removed?

File: fixed.yaml [hunk 6]
```diff
@@ -0,0 +1,37 @@
+apiVersion: apps/v1beta2
```
Comment (genevieveluyt) [inline fixed.yaml:1]: Was this file added intentionally?

File: fix.go (hunk 5 above)
Comment (dani-santos-code) [reply fix.go:1]: omg! I didn't mean to delete it! 😩I'll bring it back!

File: fixed.yaml (hunk 6 above)
Comment (dani-santos-code) [reply fixed.yaml:1]: noooo! sorry! I made a mistake when going back and forth! I should have reviewed this more carefully before asking for feedback!

Comment (dani-santos-code) [reply fixed.yaml:1]: I was supposed to have deleted the yaml and deletes the fix.go`` by accident instead!

**Review by genevieveluyt — APPROVED** [review]
LGTM!

**Review by cjones-shopify — APPROVED** [review]
This looks good to me!
